src/components/tree-section/daikin-tree-section.ts

```diff
@@ -0,0 +1,236 @@
+  disabled: boolean = false;
+
+  /**
+   * Whether the tree section is open.
+   * Treated as `false` when `disabled` is `true`.
+   */
+  @property({ type: Boolean, reflect: true })
+  open: boolean = false;
+
+  /**
+   * _Internal use._
+   * The current nesting depth when the root's children are 0.
+   * Automatically set by the parent.
+   *
+   * @private
+   */
+  @property({ type: Number, attribute: false })
+  level: number = 0;
+
+  @queryAssignedElements({ selector: "daikin-tree-section,daikin-tree-item" })
+  private readonly _children!: readonly (DaikinTreeSection | DaikinTreeItem)[];
+
+  @query("button")
+  private readonly _button!: HTMLButtonElement | null;
+
+  private get _open(): boolean {
+    return this.open && !this.disabled;
+  }
+
+  private _updateChildrenLevel(): void {
+    for (const item of this._children) {
+      item.level = this.level + 1;
+    }
+  }
+
+  private _handleClick(): void {
+    if (
+      this.dispatchEvent(
+        new Event("toggle", {
+          cancelable: true,
+        })
+      )
+    ) {
+      this.open = !this.open;
+    }
+  }
+
+  private _handleSlotChange(): void {
+    this._updateChildrenLevel();
+  }
+
+  private _handleTreeMoveFocus(event: TreeMoveFocusEvent): void {
+    handleTreeMoveFocusSection(this, event, this._children);
+  }
+
+  private _handleKeyDown(event: KeyboardEvent): void {
+    const direction = getDirectionFromKey(event.key);
+    if (!direction) {
+      return;
+    }
+
+    // Prevent scrolling.
+    event.preventDefault();
+
+    switch (direction) {
+      case "up":
+        // The item above the section header is in a previous sibling, so emit an event and let the parent handle it.
+        emitTreeMoveFocus(this, "up");
+        break;
+
+      case "down": {
+        // If the section is open, the first item under the header is the first child, so focus on that.
+        // If the section is closed or there is no child in the slot, the next sibling needs to be focused, so emit the `tree-move-focus` event and let the parent handle it.
+        const nextItem = this.open && this._children[0];
+        if (nextItem) {
+          nextItem.focus();
+        } else {
+          emitTreeMoveFocus(this, "down");
+        }
+        break;
+      }
+
+      case "left":
+        if (this._open) {
+          // Close the section if open.
+          this.open = false;
+        } else {
+          // Move focus to the parent section/root if closed.
+          emitTreeMoveFocus(this, "left");
+        }
+        break;
+
+      case "right":
+        if (!this._open) {
+          // Open the section if closed.
+          this.open = true;
+        } else {
+          // Move focus to the first child (if any) if open.
+          this._children[0]?.focus();
+        }
+        break;
+    }
+  }
+
+  override render() {
+    // eslint-disable-next-line lit-a11y/accessible-name
+    return html`<div role="treeitem" aria-selected=${this._open}>
```

Comment:
see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treeitem_role#examples

```suggestion
    return html`<div role="treeitem" aria-expanded=${this._open}>
```

Reply:
If we're providing select feature, we should also add `aria-selected` and `aria-disabled`.